### PR TITLE
Feature/1843 org vs individual edit delete post

### LINKED ISSUE #1843: Switch Profile -> Individual profile user can EDIT/DEL the post, posted as an ORG and vice-versa.
As per the requirements mentioned in PR #1436 missed to be added in PR #1767 

"user should only edit or delete what is based on that profile- Indiv or org. If indiv, I can only edit or delete posts I made as an indiv.

If org, I can only edit or delete posts made by that org.

If user has more than one org, they have to be acting as same org that created post to edit or delete."

Need to add this missing functionality for 'Organizational Switch'

related to #1849 #1855

### LINKED ISSUE #1849: Able to create post for  organization when in individual profile account and vice versa.
Environment- staging
Browser -Google chrome 
Device used-Samsung Galaxy s8
1.login with valid user name and password 
2.click on switch account. 
3.click on any organization profile. 
4.click on create post as individual. 
Expected result 
It should not allow user to create post for individual when they are in organization account.
Actual result
It allows to create post for individual when they are in organization account and vice versa. 
GIF-201117_193538

### LINKED ISSUE #1855: Able to edit individual account profile information from organization account.
Environment staging 
Browser Google chrome 
Device used-Samsung Galaxy s8
1.login with valid user name and password 
2.click switch account. 
3.select organization account. 
4.Go to helpboard. 
5.click on individual profile post and go to the profile.
6.click on pencil symbol on top right of the screen. 
7.select edit account information/edit profile/edit security.
Expected result 
Should not allow to edit individual profile account information from organization account. 
Actual result
Its allowing to edit the individual profile account information from organization account and vice versa. 
GIF-201118_120154

### PR DESCRIPTION
**Pull request by ddveloper**
### What does this pull request aim to achieve? 💯
accounts editing was enabled for user and the linked organisations, break them thus only org user can edit org account and the individual user can edit individual account; post edit/delete was also cross-enabled, now break it
_Please be concise and link any related issue(s):_
to fix #1843 , #1855 , and #1849 

- [x] I have checked that no one else is working on similar changes.
- [x] I have read the latest [**README**](README.md) and understand the development workflow.
- [x] The name of this branch is: **`feature/<branch_name>`**, or **`hotfix/<branch_name>`** if this is a hotfix, where `<branch_name>` briefly describes the issue(s) resolved and is prefixed with the issue number(s) (e.g. `1127-1130-update-git-branching-model`). The branch is created in a cloned version of this repo, **not a forked repo**.
- [x] This branch is rebased/merged with the **latest staging** (or the **latest production** if it is a hotfix).
- [x] There are no merge conflicts.
- [x] There are no console warnings and errors.
- [x] On the right hand side of this PR, I have linked the appropriate issue(s) under "Issues" and added a project progress under "Projects".
- [x] The title describes the issue(s) being addressed using format: **`<issue_numbers> - <brief_issue_description>`** (e.g. `#1127/#1130 - Update Git Branching Model`)

### CODE REVIEW

File: client/src/components/CreatePost/CreatePost.js [hunk 1]
```diff
@@ -209,8 +112,6 @@ const CreatePost = ({ onCancel, loadPosts, ...props }) => {
       value={{ form, setForm, currentStep, setCurrentStep, postId, setPostId }}
     >
       <Wrapper onCancel={clearState} {...props}>
-        <Step1 gtmPrefix={props.gtmPrefix} />
-        <Step2 user={props.user} gtmPrefix={props.gtmPrefix} />
         <Step4 gtmPrefix={props.gtmPrefix} />
       </Wrapper>
       <Step3 onCancel={clearState} {...props} gtmPrefix={props.gtmPrefix} />
```
Comment (Zaydme) [inline client/src/components/CreatePost/CreatePost.js]: These can be renamed to step 1 and 2, to not cause confusion in the future.

Comment (ddveloper) [reply client/src/components/CreatePost/CreatePost.js]: 👍  will do

**Review by rabbi08 — CHANGES_REQUESTED** [review]
- EDIT post, posted as other user on Help Board: PASS
- EDIT post, posted as other user on View Profile: PASS
- EDIT post, posted as other user on POST page: PASS

- EDIT post, posted as other user on EDIT post page: FAIL, 

Steps: Open the post as EDIT and EDIT the post-> switch profile on same page -> YEs -> EDIT pop up appears -> and user can EDIT. this happen when user EDIT post and switch.
Refer below GIF
ezgif com-gif-maker (64)

**Comment by ddveloper** [comment]
Hi @rabbi08 , please have a check : )

**Comment by Zaydme** [comment]
@ddveloper can you merge the last staging into this? I wonder if this is something we need to worry about https://github.com/FightPandemics/FightPandemics/pull/1480#discussion_r528045060

**Comment by rabbi08** [comment]
Verified in feature build:
- The post is created with the selected user, No option pop up which 'Create post' is there anymore.
- Only the selected user can edit their post .

Thanks @ddveloper

**Review by rabbi08 — APPROVED** [review]

**Comment by Zaydme** [comment]
@ddveloper there was some conflicts with the message button, since it needs  the old isOwner and ownUser, so orgs can't contact owner and vice versa, so I renamed yours to isSelf, and kept the old isOwner and ownUser.
Take a look and merge if looks good.

**Comment by ddveloper** [comment]
> @ddveloper there was some conflicts with the message button, since it needs the old isOwner and ownUser, so orgs can't contact owner and vice versa, so I renamed yours to isSelf, and kept the oldisOwner and ownUser

Yes @Zaydme , that should be fine

**Review by Zaydme — APPROVED** [review]

**Comment by nivetheatha** [comment]
Tested in staging 
Works as expected .

**Pull request merged**